pkg/R/sparkR.R

```diff
@@ -149,11 +156,14 @@ sparkR.init <- function(
     }
     f <- file(path, open='rb')
     sparkRBackendPort <- readInt(f)
+    monitorPort <- readInt(f)
     close(f)
     file.remove(path)
-    if (length(sparkRBackendPort) == 0) {
+    if (length(sparkRBackendPort) == 0 || sparkRBackendPort == 0 ||
+        length(monitorPort) == 0 || monitorPort == 0) {
       stop("JVM failed to launch")
     }
+    assign(".monitorConn", socketConnection(port = monitorPort), envir = .sparkREnv)
```

Comment:
I can see that point, but the problem is `reg.finalizer` doesn't work in all cases. For example if you shutdown R and choose to save the workspace (or if you exit RStudio) then the socket connections get closed before `reg.finalizer` is run. So in those cases the JVM will shutdown before we get a chance to call `sc.stop`.  

I think the right thing to do might be to update our documentation and educate users that they should call `sparkR.stop` at the end of their program